bob.bash

```diff
@@ -43,6 +43,10 @@ python "${BOB_DIR}/config_system/update_config.py" \
        --bob-config "${BUILDDIR}/config.json" \
        ${BOB_CONFIG_OPTS}
 
+# Get a hash of the environment so we can detect if we need to
+# regenerate the build.ninja
+python "${BOB_DIR}/scripts/env_hash.py" "${BUILDDIR}/.env.hash"
```

Comment:
Probably for a follow-up commit, but we ought to consider whether we need to do this in the config system wrappers too, as Soong might need this (which doesn't run `buildme`/`bob.bash`).

Reply:
I think we're ok for soong. Android does an environment check and I believe it regenerates build.ninja if it sees it change. For us it would mean the bob modules are also reprocessed. See soong/cmd/soong_env/soong_env.go

Reply:
OK, in order for that to work for us, our plugin needs to call android.Getenv(), and android will record the dependency on that environment variable.